cli/src/commands/workspace/update_stale.rs

```diff
@@ -0,0 +1,170 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use jj_lib::object_id::ObjectId;
+use jj_lib::op_store::OpStoreError;
+use jj_lib::operation::Operation;
+use jj_lib::repo::{ReadonlyRepo, Repo};
+use tracing::instrument;
+
+use crate::cli_util::{
+    check_stale_working_copy, print_checkout_stats, short_commit_hash, CommandHelper,
+    WorkingCopyFreshness, WorkspaceCommandHelper,
+};
+use crate::command_error::{internal_error_with_message, user_error, CommandError};
+use crate::ui::Ui;
+
+/// Update a workspace that has become stale
+///
+/// For information about stale working copies, see
+/// https://github.com/martinvonz/jj/blob/main/docs/working-copy.md.
+#[derive(clap::Args, Clone, Debug)]
+pub struct WorkspaceUpdateStaleArgs {}
+
+fn create_and_check_out_recovery_commit(
+    ui: &mut Ui,
+    command: &CommandHelper,
+) -> Result<Arc<ReadonlyRepo>, CommandError> {
+    let mut workspace_command = command.workspace_helper_no_snapshot(ui)?;
+    let workspace_id = workspace_command.workspace_id().clone();
+    let mut tx = workspace_command.start_transaction().into_inner();
+
+    let (mut locked_workspace, commit) =
+        workspace_command.unchecked_start_working_copy_mutation()?;
+    let commit_id = commit.id();
+
+    let mut_repo = tx.mut_repo();
+    let new_commit = mut_repo
+        .new_commit(
+            command.settings(),
+            vec![commit_id.clone()],
+            commit.tree_id().clone(),
+        )
+        .write()?;
+    mut_repo.set_wc_commit(workspace_id, new_commit.id().clone())?;
+    let repo = tx.commit("recovery commit");
+
+    locked_workspace.locked_wc().recover(&new_commit)?;
+    locked_workspace.finish(repo.op_id().clone())?;
+
+    writeln!(
+        ui.status(),
+        "Created and checked out recovery commit {}",
+        short_commit_hash(new_commit.id())
+    )?;
+
+    Ok(repo)
+}
+
+/// Loads workspace that will diverge from the last working-copy operation.
+fn for_stale_working_copy(
+    ui: &mut Ui,
+    command: &CommandHelper,
+) -> Result<(WorkspaceCommandHelper, bool), CommandError> {
+    let workspace = command.load_workspace()?;
+    let op_store = workspace.repo_loader().op_store();
+    let (repo, recovered) = {
+        let op_id = workspace.working_copy().operation_id();
+        match op_store.read_operation(op_id) {
+            Ok(op_data) => (
+                workspace.repo_loader().load_at(&Operation::new(
+                    op_store.clone(),
+                    op_id.clone(),
+                    op_data,
+                ))?,
+                false,
+            ),
+            Err(e @ OpStoreError::ObjectNotFound { .. }) => {
+                writeln!(
+                    ui.status(),
+                    "Failed to read working copy's current operation; attempting recovery. Error \
+                     message from read attempt: {e}"
```

Comment:
Nit: Use a raw string here.

Reply:
Hmm, I'm okay with either way, but if we don't like the default string wrapping, maybe we should disable it instead of abusing raw string?

Reply:
I mildly prefer keeping this as-is or `indoc`-ed raw string to a raw string which breaks indentation.

I also agree with Yuya's point (if I understand the situation correctly): if we want to reconfigure `cargo fmt`, we should do it separately.

Reply:
> Hmm, I'm okay with either way, but if we don't like the default string wrapping, maybe we should disable it instead of abusing raw string?

I don't really care about string wrapping and just want to prevent some human errors with raw strings. You can leave it as is if you prefer that.

Reply:
Ok, I'll leave it unchanged as this PR just moves codes around. 

FWIW, I don't think raw string is strictly better because it disables backslash escapes.